source/controller/src/map_ctrl_utils.c

```diff
@@ -115,11 +115,11 @@ static void check_add_op_class_channel(map_op_class_list_t *merged_list, map_op_
         find_op_class->op_class = op_class->op_class;
         find_op_class->pref     = pref;
         find_op_class->reason   = 0; /* TODO... */
-    }
 
-    /* Add channel */
-    if (!map_is_channel_in_op_class_list(find_op_class, channel) && find_op_class->channel_count < MAX_CHANNEL_PER_OP_CLASS) {
-        find_op_class->channel_list[find_op_class->channel_count++] = channel;
+        /* Add channel */
+        if (!map_is_channel_in_op_class_list(find_op_class, channel) && find_op_class->channel_count < MAX_CHANNEL_PER_OP_CLASS) {
+            find_op_class->channel_list[find_op_class->channel_count++] = channel;
+        }
     }
```

Comment:
This is wrong, if you want to eliminate warning set find_op_class to NULL at definition.